source/distributed/tria_base.cc

```diff
@@ -362,36 +362,125 @@ namespace parallel
   TriangulationBase<dim, spacedim>::compute_vertices_with_ghost_neighbors()
     const
   {
-    // TODO: we are not treating periodic neighbors correctly here. If we do
-    // we can remove the overriding implementation for p::d::Triangulation
-    // that is currently using a p4est callback to get correct ghost neighbors
-    // over periodic faces.
-    Assert(this->get_periodic_face_map().size() == 0, ExcNotImplemented());
+    // 1) collect for each vertex on periodic faces all vertices it coincides
+    //    with
+    std::map<unsigned int, std::set<unsigned int>>
+      vertex_to_coinciding_vertices;
+    {
+      // 1a) collect nodes coinciding due to periodicity
+      std::map<unsigned int, unsigned int> vertex_to_coinciding_vertex;
+      for (auto &cell : this->active_cell_iterators())
+        if (cell->is_locally_owned() || cell->is_ghost())
+          for (auto i = 0u; i < GeometryInfo<dim>::faces_per_cell; ++i)
```

Comment:
```suggestion
          for (unsigned int i = 0; i < GeometryInfo<dim>::faces_per_cell; ++i)
```
(let us spell out integers consistent with the rest of the library)